src/devices/src/legacy/rtc_pl031.rs

```diff
@@ -1,336 +1,73 @@
-// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
-// SPDX-License-Identifier: Apache-2.0
+use std::{convert::TryInto, sync::Arc};
 
-//! ARM PL031 Real Time Clock
-//!
-//! This module implements a PL031 Real Time Clock (RTC) that provides to provides long time base counter.
-//! This is achieved by generating an interrupt signal after counting for a programmed number of cycles of
-//! a real-time clock input.
-//!
-use std::{fmt, result, time::Instant};
+use logger::{metrics::RTCDeviceMetrics, warn};
+use vm_superio::RTC;
 
 use crate::BusDevice;
-use logger::{warn, IncMetric, METRICS};
-use utils::byte_order;
 
-// As you can see in https://static.docs.arm.com/ddi0224/c/real_time_clock_pl031_r1p3_technical_reference_manual_DDI0224C.pdf
-// at section 3.2 Summary of RTC registers, the total size occupied by this device is 0x000 -> 0xFFC + 4 = 0x1000.
-// From 0x0 to 0x1C we have following registers:
-const RTCDR: u64 = 0x0; // Data Register.
-const RTCMR: u64 = 0x4; // Match Register.
-const RTCLR: u64 = 0x8; // Load Regiser.
-const RTCCR: u64 = 0xc; // Control Register.
-const RTCIMSC: u64 = 0x10; // Interrupt Mask Set or Clear Register.
-const RTCRIS: u64 = 0x14; // Raw Interrupt Status.
-const RTCMIS: u64 = 0x18; // Masked Interrupt Status.
-const RTCICR: u64 = 0x1c; // Interrupt Clear Register.
-                          // From 0x020 to 0xFDC => reserved space.
-                          // From 0xFE0 to 0x1000 => Peripheral and PrimeCell Identification Registers which are Read Only registers.
-                          // AMBA standard devices have CIDs (Cell IDs) and PIDs (Peripheral IDs). The linux kernel will look for these in order to assert the identity
-                          // of these devices (i.e look at the `amba_device_try_add` function).
-                          // We are putting the expected values (look at 'Reset value' column from above mentioned document) in an array.
-const PL031_ID: [u8; 8] = [0x31, 0x10, 0x04, 0x00, 0x0d, 0xf0, 0x05, 0xb1];
-// We are only interested in the margins.
-const AMBA_ID_LOW: u64 = 0xFE0;
-const AMBA_ID_HIGH: u64 = 0x1000;
-
-pub enum Error {
-    BadOffset(u64, &'static str),
-}
-
-impl fmt::Display for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match self {
-            Error::BadOffset(offset, ops) => write!(f, "Bad {} offset: {}", ops, offset),
-        }
-    }
-}
-type Result<T> = result::Result<T, Error>;
-
-/// A RTC device following the PL031 specification..
-pub struct Rtc {
-    previous_now: Instant,
-    tick_offset: i64,
-    // This is used for implementing the RTC alarm. However, in Firecracker we do not need it.
-    match_value: u32,
-    // Writes to this register load an update value into the RTC.
-    load: u32,
-    imsc: u32,
-    ris: u32,
-}
-
-impl Rtc {
-    /// Constructs an AMBA PL031 RTC device.
```

Comment:
I would keep this comment in the Firecracker codebase, even if the implementation lives elsewhere.